subscriptions-api-openapi.yaml

```diff
@@ -0,0 +1,206 @@
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: "#/components/schemas/Subscription"
+                    description: the subscription that was requested with teh 'id' parameter
+                  - type: array
+                    description: a list of subscriptions
+                    items:
+                       $ref: "#/components/schemas/Subscription"
+    put:
+      description: Update a subscription
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Subscription"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Subscription"
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Subscription"
+      responses:
+        "200":
+          description: Updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Subscription"
+        "201":
+          description: Created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Subscription"
+          headers:
+            Location:
+              schema:
+                type: string
+                format: url
+        "400":
+          description: Invalid or malformed request
+    delete:
+      description: Delete a subscription
+      parameters:
+        - in: "query"
+          name: "id"
+          required: true
+          schema:
+            type: string
+          description: The id of an existing subscription
+      responses:
+        "200":
+          description: Successfully deleted
+        "400":
+          description: Invalid or malformed request
+        "404":
+          description: Subscription not found
+    options:
+      description: Discover supported features and methods for this endpoint
+      responses:
+        "200":
+          description: OK
+          headers:
+            Allow:
+              schema:
+                type: string
+                default: "GET,PUT,POST,DELETE,OPTIONS"
+components:
+  schemas:
+    Subscription:
+      properties:
+        id:
+          type: string
+          description: REQUIRED. The unique identifier of the subscription in the scope of the subscription manager.
+          example: 1119920371
+        protocol:
+          type: string
+          enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
+          description: REQUIRED. Identifier of a delivery protocol.
+          example: "HTTP"
+        protocolsettings:
+          type: object
+          anyOf:
+            - $ref: "#/components/schemas/HTTPSettings"
```

Comment:
same concern - we can't possibly list all configuration setting for all protocols.

in discovery this was left as a map to specify necessary subscription parameters

Reply:
We have a set of protocols that we have a clearly considered set of settings for as a baseline. It's better to have a baseline than leave it completely open.